.github/workflows/pull-request-checks.yml

```
@@ -275,12 +279,41 @@ jobs:
         with:
           command: audit
 
+  # Only run check if the project still compiles on MSRV, test are run on stable
+  check-msrv:
+    name: Run `cargo check` on MSRV
+    runs-on: Ubuntu-20.04
+    needs: changes
+    if: ${{ needs.changes.outputs.rust == 'true' }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Retrieve MSRV from workspace Cargo.toml
+        id: rust_version
+        uses: SebRollen/toml-action@v1.0.2
+        with:
+          file: Cargo.toml
+          field: "workspace.package.rust-version"
+
+      - name: Enable rust toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ steps.rust_version.outputs.value }}
+
+      - name: Enable cache
+        uses: Swatinem/rust-cache@v2.4.0
+
+      - name: Run `cargo check`
+        run: cargo check --all-targets --all-features
+
   cargo-test:
     name: Run cargo test
     runs-on: Ubuntu-20.04
     env:
       RUSTFLAGS: -D warnings
-    needs: [cargo-fmt, cargo-clippy]
+    needs: changes
 
```

I would add this condition as running `cargo test` for no Rust code change is useless.
```suggestion
    if: ${{ needs.changes.outputs.rust == 'true' || needs.changes.outputs.workflows == 'true' }}
```